validator/likecoin-chain-node/setup-a-node.md

```diff
@@ -1,108 +1,440 @@
+
+Then, we will need to obtain the latest block height and hash from one of the rpc endpoints:
+
+```shell
+curl -s https://fotan-node-1.like.co:443/rpc/block | jq -r '"height: \(.result.block.header.height)\nhash: \(.result.block_id.hash)"'
+```
+
+Next, we will edit `~/.liked/config/config.toml`. Locate the `[statesync]` section and update it to the following values:
+
+```toml
 [statesync]
 enable = true
 rpc_servers = "https://fotan-node-1.like.co:443/rpc/,https://fotan-node-2.like.co:443/rpc/"
-trust_height = 225500
-trust_hash = "10737475ED545C9BE1A48E1F99BD8D2941E7E286EDB71FC1B9E15E27032144DB"
+trust_height = <known height>
+trust_hash = "<known hash>"
 trust_period = "168h0m0s"
+discovery_time = "15s"
+temp_dir = ""
+chunk_request_timeout = "60s"
+chunk_fetchers = "4"
+```
+
+Then, the node will use state sync to synchronize with the network instead of replaying from genesis at launch.
+
+If you have already started the service once, i.e. there is an ongoing sync, you will have to reset the state back to genesis, otherwise state sync will not start:
+
+```shell
+~/liked unsafe-reset-all
 ```
 
-### Becoming a validator
+### Change node configuration
+
+You are recommended to review `~/.liked/config/config.toml` and `~/.liked/config/app.toml`, then change the defaults as desired.
+
+In particular, please set a minimum gas price for your node and enable state sync in `app.toml`:
+
+```toml
+minimum-gas-prices = "1.0nanolike"
+snapshot-interval = 1000
+```
 
-For a validator node, you will need a node synchronized, which you may setup following the steps above.
+This avoids spams on the network and help new nodes sync to the latest state quickly.
+
+For more details, please refer to [Node Configuration docs](https://docs.like.co/validator/likecoin-chain-node/node-configuration).
+
+### Activate the service
+
+Now we will call the setup script again to activate the systemd config and start the node.
+
+The service is configured to start the node via Cosmovisor. It will auto-start whenever the machine is booted.
+
+The service file will be placed at `/etc/systemd/system/liked.service`.
+
+> For migrating validators, please ensure the old instance is turned off before you proceed!
+
+```shell
+cd ~/likecoin-chain
+make -C deploy/cmd/deploy/assets initialize-systemctl
+make -C deploy/cmd/deploy/assets start-node
+```
 
-You also need some LikeCoin in your operator address.
+New nodes will now synchronize with the network. We can continue below steps to create a validator afterwards.
 
-Then you can use the following command to create a validator, turning your node into a validator node:
+You can check the sync status by:
 
-`docker-compose run --rm create-validator \`\
-&#x20;   `--amount <AMOUNT> \`\
-&#x20;   `--details <DETAILS> \`\
-&#x20;   `--commission-rate <COMMISSION_RATE>`
+```shell
+curl -s localhost:26657/status | jq -r '"synced height: \(.result.sync_info.latest_block_height)\ncatching up: \(.result.sync_info.catching_up)"'
```

Comment:
I recommend drop `jq`. It need to install separately.

Reply:
but this is way more clear than instructing user to read a long json response; how about add a section to install make and jq?

Reply:
The default response of `curl -s localhost:26657/status` is indented json. Like the following:


```
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "node_info": {
      "protocol_version": {
        "p2p": "8",
        "block": "11",
        "app": "0"
      },
      "id": "89275a03d99f57bcdc9a192d9358d6836839412b",
      "listen_addr": "tcp://14.199.56.25:26656",
      "network": "likecoin-mainnet-2",
      "version": "0.34.14",
      "channels": "40202122233038606100",
      "moniker": "pn51",
      "other": {
        "tx_index": "on",
        "rpc_address": "tcp://0.0.0.0:26657"
      }
    },
    "sync_info": {
      "latest_block_hash": "1AC8E0C3AD205BB11CE6632268114D45417A480197198D143C278B83D8F59790",
      "latest_app_hash": "5B5588BC7F1F771776FDE14C64B7F97BE56A381F88F1C728C5CE4B57B32D455A",
      "latest_block_height": "2969322",
      "latest_block_time": "2022-03-15T08:22:22.201896873Z",
      "earliest_block_hash": "9DEFBA987C2D0ACBD6B9E3A8311651F0043F3847A49D56C5AA047CDC247AE2E7",
      "earliest_app_hash": "E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855",
      "earliest_block_height": "1",
      "earliest_block_time": "2021-08-18T13:30:00Z",
      "catching_up": false
    },
    "validator_info": {
      "address": "060C1E1EA44FEC3A3AFAEA795D706673D5C0E0C8",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "o5qa+Rr0U2y7qR+ytiDQdUDviETXDm2pNJyPF6g1wFA="
      },
      "voting_power": "0"
    }
  }
}
```

Reply:
removed jq usage